src/include/sof/spi.h

```diff
@@ -0,0 +1,188 @@
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Zhigang.wu <zhigang.wu@linux.intel.com>
+ */
+
+#ifndef __INCLUDE_SPI__
+#define __INCLUDE_SPI__
+
+#include <stdint.h>
+
+#define DMA_HANDSHAKE_SSPI_TX		26
+#define DMA_HANDSHAKE_SSPI_RX		27
+
+#define SSI_SLAVE_CLOCK_CTL		0x00081C60
+
+enum spi_type {
+	SPI_TYPE_INTEL_RECEIVE  = 0,
+	SPI_TYPE_INTEL_TRANSMIT,
+};
+
+enum {
+	SPI_TRIGGER_START = 0,
+	SPI_TRIGGER_STOP,
+};
+
+enum sof_spi_type {
+	SOF_SPI_INTEL_SLAVE = 0,
+	SOF_SPI_INTEL_MASTER,
+};
+
+/* CTRLR0 */
+/* 00-standard spi; 01-dual spi; 10-quad spi */
+#define SPI_FRAME_FORMAT(x)	((x) << 21)
+#define SPI_DATA_FRAME_SIZE(x)	((x) << 16)
+/* 0-slave txt enabled; 1-slave txt disabled */
+#define SPI_SLV_OE(x)		((x) << 10)
+/* 00-both; 01-transmit only;
+ * 10-receive only; 11-eeprom read
+ */
+#define SPI_TRANSFER_MODE(x)	((x) << 8)
+/* 0-inactive low; 1-inactive high */
+#define SPI_SCPOL(x)		((x) << 7)
+/* 0-first edge capture; 1-one cycle after cs line */
+#define SPI_SCPH(x)		((x) << 6)
+/* 00-moto spi; 01-ti ssp; 10-ns microwire */
+#define SPI_FRAME_TYPE(x)	((x) << 4)
+
+/* SSIENR */
+#define SPI_SSIEN		1
+
+/* IMR */
+/* 0-masked; 1-unmasked;
+ * receive FIFO full interrupt mask/unmask
+ */
+#define SPI_IMR_RXFIM(x)	((x) << 4)
+/* 0-masked; 1-unmasked;
+ * receive FIFO overflow interrupt mask/unmask
+ */
+#define SPI_IMR_RXOIM(x)	((x) << 3)
+/* 0-masked; 1-unmasked;
+ * transmit FIFO overflow interrupt mask/unmask
+ */
+#define SPI_IMR_TXOIM(x)	((x) << 1)
+/* 0-masked; 1-unmasked;
+ * transmit FIFO empty interrupt mask/unmask
+ */
+#define SPI_IMR_TXEIM(x)	((x) << 0)
+
+/* DMACR */
+/* 0-transmit DMA disable; 1-transmit DMA enable */
+#define	SPI_DMACR_TDMAE(x)	((x) << 1)
+/* 0-receive DMA disable; 1-receive DMA enable */
+#define SPI_DMACR_RDMAE(x)	((x) << 0)
+/* DMATDLR/DMARDLR */
+/* transmit data level: 0~255 */
+#define SPI_DMATDLR(x)		((x) << 0)
+/* receive data level: 0~255 */
+#define SPI_DMARDLR(x)		((x) << 0)
+
+/* spi register */
+#define SPI_SLAVE_BASE	0x00080000
```

Comment:
This should be in memory.h as it could be different on other platforms.

Reply:
@lgirdwood 
will move it to memory.h file.